### PR TITLE
feat: publish automatically when item is validated

### PR DESCRIPTION
**Pull request by ReidyT**
Publish the item after the validation if it is valid.
If the item is private, it will be set to public after the validation succeeded (the frontend has to ask to the user).

### CODE REVIEW

File: src/services/item/plugins/published/service.ts [hunk 1]
```diff
@@ -123,7 +128,16 @@ export class ItemPublishedService {
...
-    await itemTagRepository.getType(item, ItemTagType.Public, { shouldThrow: true });
+    const tag = await itemTagRepository.getType(item, ItemTagType.Public, {
+      shouldThrow: shouldBePublic,
+    });
+
+    // if the item can be private and be published, set it to public automatically.
+    // it's usefull for the publishing automatically after the validation.
+    // the user is asked to set the item to public in the frontend.
+    if (!tag && !shouldBePublic) {
+      await itemTagRepository.post(actor, item, ItemTagType.Public);
```
Comment (ReidyT) [inline src/services/item/plugins/published/service.ts:139]: It is the responsibility of the publish service or the validation service to set the item to public if it is private ?

File: src/services/item/plugins/validation/service.ts [hunk 2]
```diff
@@ -31,12 +32,19 @@ import { stripHtml } from './utils';
...
   itemService: ItemService;
+  private publishService: ItemPublishedService;
   fileService: FileService;
   imageClassifierApi?: string;
 
-  constructor(itemService: ItemService, fileService: FileService, imageClassifierApi?: string) {
+  constructor(
+    itemService: ItemService,
+    fileService: FileService,
+    publishService: ItemPublishedService,
```
Comment (ReidyT) [inline src/services/item/plugins/validation/service.ts:42]: For now I inject the dependency manually, I will check if it is a small change to use a DI library for that.

File: src/services/item/plugins/published/service.ts [hunk 3]
```diff
@@ -114,7 +114,12 @@ export class ItemPublishedService {
     };
   }
 
-  async post(actor: Actor, repositories: Repositories, itemId: string) {
+  async post(
+    actor: Actor,
+    repositories: Repositories,
+    itemId: string,
+    { shouldBePublic = true }: { shouldBePublic: boolean } = { shouldBePublic: true },
```
Comment (spaenleh) [inline src/services/item/plugins/published/service.ts:121]: Maybe handling the missing option inside the function will be less cumbersome ?

**Review by spaenleh — APPROVED** [review]
This looks good to me.

**Pull request merged**